### PR TITLE
ci(jenkins): enable lint commit

### PR DESCRIPTION
**Pull request by v1v**
### What

Enable git commit linting:
- locally
- CI Jenkins at the very beginning (linting stage)

Further details:
- https://github.com/conventional-changelog/commitlint
- https://github.com/conventional-changelog/commitlint/tree/master/@commitlint/config-lerna-scopes

### Why

To help with the changelog

### Tests

- Locally if commit doesn't match the allowed format

```
git commit -m "test something blabla"
husky > pre-commit (node v12.10.0)
No staged files match any of provided globs.
husky > commit-msg (node v12.10.0)
⧗   input: test something blabla
✖   subject may not be empty [subject-empty]
✖   type may not be empty [type-empty]

✖   found 2 problems, 0 warnings
ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint

husky > commit-msg hook failed (add --no-verify to bypass)
```
- Locally if the commit follows the format:

```
git commit -m "ci(jenkins): enable lint commit script for the CI"
husky > pre-commit (node v12.10.0)
No staged files match any of provided globs.
husky > commit-msg (node v12.10.0)
[feature/lint-commit 705d94a] ci(jenkins): enable lint commit script for the CI
 2 files changed, 32 insertions(+), 1 deletion(-)
 create mode 100755 scripts/lint-commits.sh
```

- CI if right commit format

image

- CI if commit format is not right

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
     "github-release": "node scripts/github-release",
     "build-docs": "PREVIEW=1 sh ./scripts/build_docs.sh apm-agent-rum-js ./docs ./build",
     "lint": "run-p --silent lint:*",
+    "lint:commit": "scripts/lint-commits.sh",
```
Comment (vigneshshanmugam) [inline package.json]: Its better to not do this here instead move CI related things to `ci/scripts/lint.sh`, This script is not so useful for dev workflow.

Comment (v1v) [inline package.json]:
```suggestion
    "lint:commit": "scripts/lint-commits.sh",
```

File: commitlint.config.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+module.exports = { extends: ['@commitlint/config-lerna-scopes'] }
```
Comment (vigneshshanmugam) [inline commitlint.config.js]: This file can be removed in favor of adding the `@commitlint/config-lerna-scopes` to `commitlint` config in `package.json`

File: .ci/scripts/lint.sh [hunk 3]
```diff
@@ -1,3 +1,4 @@
 #!/usr/bin/env bash
 npm install
 npm run lint
+npm run lint:commit
```
Comment (v1v) [inline .ci/scripts/lint.sh]:
```suggestion
scripts/lint-commits.sh
```

File: package.json [hunk 4]
```diff
@@ -146,5 +150,11 @@
     "@elastic/apm-rum-core": "file:packages/rum-core",
     "@elastic/apm-rum-react": "file:packages/rum-react",
     "@elastic/apm-rum-vue": "file:packages/rum-vue"
+  },
+  "commitlint": {
```
Comment (vigneshshanmugam) [inline package.json]: nit: can be moved near the husky and lint-staged params.

**Pull request merged**